### PR TITLE
Kernel heap & ktime `Instant` modification

### PR DESCRIPTION
**Pull request by js2xxx**
- Added kernel heap allocator
- Changed `ktime::Instant`'s representation from nanoseconds to microseconds.

### CODE REVIEW

File: mizu/kernel/Makefile [hunk 1]
```diff
@@ -7,7 +7,9 @@ KERNEL_ASM  := $(DEBUG_DIR)/$(KERNEL).asm
 
 CARGO_ARGS := --target $(TARGET) \
 	--no-default-features \
-	--features $(BOARD)
+	--features $(BOARD) \
+	-Zbuild-std=core,compiler_builtins,alloc,panic_abort \
```
Comment (js2xxx) [inline mizu/kernel/Makefile:11]: Moved `build-std` from config.toml to Makefile to limit it only in production builds, with unit tests getting rid of it.

File: mizu/kernel/link.ld [hunk 2]
```diff
@@ -51,6 +51,10 @@ SECTIONS
         _sstack = ALIGN(4K);
         . += _stack_size * _max_harts;
         _estack = ALIGN(8);
+
+        _sheap = ALIGN(4K);
+        . += _heap_size;
+        _eheap = ALIGN(8);
```
Comment (js2xxx) [inline mizu/kernel/link.ld:57]: Added heap space in the symbol table. Configurable in the future.

File: mizu/lib/config/src/qemu-virt.rs [hunk 3]
```diff
@@ -2,4 +2,5 @@ use num_rational::Ratio;
 
 pub const KERNEL_START: usize = 0x80200000;
 
-pub const TIME_FREQ: Ratio<u128> = Ratio::new_raw(1, 12_500_000);
+pub const TIME_FREQ: u128 = 12_500_000;
+pub const TIME_FREQ_M: Ratio<u128> = Ratio::new_raw(2, 25); // 10^6 / FREQ
```
Comment (js2xxx) [inline mizu/lib/config/src/qemu-virt.rs:6]: Adjusted manually to avoid overflow.

**Review by js2xxx — COMMENTED** [review]
Good.

**Review by IndifferentArea — APPROVED** [review]
good job.

**Pull request merged**